lib/models/room.js

```diff
@@ -386,32 +575,41 @@ Room.prototype.addEvents = function(events, duplicateStrategy) {
             }
             // TODO: We should have a filter to say "only add state event
             // types X Y Z to the timeline".
-            this.addEventsToTimeline([events[i]]);
+            this._addLiveEvents([events[i]]);
         }
     }
 };
 
 /**
  * Removes events from this room.
- * @param {String} event_ids A list of event_ids to remove.
+ * @param {String[]} event_ids A list of event_ids to remove.
  */
 Room.prototype.removeEvents = function(event_ids) {
-    // avoids defining a function in the loop, which is a lint error
-    function removeEventWithId(timeline, id) {
-        // NB. we supply reverse to search from the end,
-        // on the assumption that recents events are much
-        // more likley to be removed than older ones.
-        return utils.removeElement(timeline, function(e) {
-            return e.getId() == id;
-        }, true);
+    for (var i = 0; i < event_ids.length; ++i) {
+        this.removeEvent(event_ids[i]);
     }
+};
 
-    for (var i = 0; i < event_ids.length; ++i) {
-        var removed = removeEventWithId(this.timeline, event_ids[i]);
-        if (removed) {
-            this.emit("Room.timeline", removed, this, undefined, true);
-        }
+/**
+ * Removes a single event from this room.
+ * @param {String} eventId  The id of the event to remove
+ * @return {?MatrixEvent} the removed event, or null if none
```

Comment:
What does `or null if none` mean? Do you mean "or null if no event was removed"? We're returning a `boolean` on `:601` though.